react/week4/src/components/GetUsersData.js

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useContext } from "react";
+import { userListContext } from "../App";
+import SearchLoader from "./SearchLoader";
+import UserList from "./UserList";
+
+function GetUsersData() {
+  const { setUser, input, error, setError, loading, setLoading } = useContext(
+    userListContext
+  );
+  useEffect(() => {
+    const SEARCH_URL = `https://api.github.com/search/users?q=${input}`;
+    (async () => {
+      try {
+        const userData = await fetch(SEARCH_URL);
+        const users = await userData.json();
+        setUser(users);
+        setLoading(!loading);

```

Comment:
I would move this to a `finally {}` block (after catch) so we set `loading = false` even if the request succeeds or fails

Reply:
corrected